Jenkinsfile

```diff
@@ -14,52 +14,41 @@ node {
 
 def runStages() {
 
-    label = "test-${UUID.randomUUID().toString()}"
+    openShift.withNode(yaml: "openshift/Jenkins/slave_pod_template.yaml") {
 
-    podTemplate(
-        label: label,
-        slaveConnectTimeout: 120,
-        serviceAccount: pipelineVars.jenkinsSvcAccount,
-        cloud: "openshift",
-        namespace: pipelineVars.defaultNameSpace,
-        yaml: readTrusted("openshift/Jenkins/slave_pod_template.yaml")
-    ) {
-        node(label) {
+        checkout scm
 
-            checkout scm
+        changedFiles = changedFiles()
 
-            changedFiles = changedFiles()
-
-            stage("Bundle install") {
-                runBundleInstall()
-            }
+        stage("Bundle install") {
+            runBundleInstall()
+        }
 
-            stage("Prepare the db") {
-                withStatusContext.dbMigrate {
-                    sh "cp config/database.yml.example config/database.yml"
-                    sh "bundle exec rake db:migrate --trace"
-                    sh "bundle exec rake db:test:prepare"
-                }
+        stage("Prepare the db") {
+            withStatusContext.dbMigrate {
+                sh "cp config/database.yml.example config/database.yml"
+                sh "bundle exec rake db:migrate --trace"
+                sh "bundle exec rake db:test:prepare"
             }
+        }
 
-            stage("Unit tests") {
-                withCredentials([string(credentialsId: "codecov_token", variable: "CODECOV_TOKEN")]) {
-                    withStatusContext.unitTest {
-                        sh "bundle exec rake test:validate"
-                    }
+        stage("Unit tests") {
+            withCredentials([string(credentialsId: "codecov_token", variable: "CODECOV_TOKEN")]) {
+                withStatusContext.unitTest {
+                    sh "bundle exec rake test:validate"
                 }
             }
+        }
 
-            if (currentBuild.currentResult == "SUCCESS" &&
-                env.BRANCH_NAME == "master" &&
-                ("Gemfile" in changedFiles || "openshift/Jenkins/Dockerfile" in changedFiles)) {
-                // If Gemfiles or Jenknis slave's Dockerfile changed we need to rebuild the jenkins slave image
-                stage("Rebuild the image") {
-                    openshiftBuild(
-                        bldCfg: "jenkins-slave-base-centos7-ruby25-openscap",
-                        namespace: "jenkins",
-                    )
-                }
+        if (currentBuild.currentResult == "SUCCESS" &&
+            env.BRANCH_NAME == "master" &&
+            ("Gemfile.lock" in changedFiles || "Gemfile" in changedFiles || "openshift/Jenkins/Dockerfile" in changedFiles)) {
```

Comment:
I'm not sure we should be committing the Gemfile.lock, but for now, I guess this is fine.

Reply:
We need to rebuild jenkins slave image to update gem cache